data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetEth1Data.java

```diff
@@ -0,0 +1,102 @@
+import tech.pegasys.teku.api.ChainDataProvider;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.response.v1.teku.GetEth1DataResponse;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.AsyncApiResponse;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.spec.datastructures.blocks.Eth1Data;
+import tech.pegasys.teku.spec.datastructures.metadata.StateAndMetaData;
+import tech.pegasys.teku.validator.coordinator.Eth1DataCache;
+
+public class GetEth1Data extends MigratingEndpointAdapter {
+
+  public static final String ROUTE = "/teku/v1/beacon/state/eth1data";
+
+  private static final SerializableTypeDefinition<Eth1Data> ETH1DATA_RESPONSE_TYPE =
+      SerializableTypeDefinition.<Eth1Data>object()
+          .name("GetEth1DataResponse")
+          .withField("data", Eth1Data.SSZ_SCHEMA.getJsonTypeDefinition(), Function.identity())
+          .build();
+
+  private final ChainDataProvider chainDataProvider;
+  private final Eth1DataCache eth1DataCache;
+
+  public GetEth1Data(final DataProvider dataProvider, final Eth1DataCache eth1DataCache) {
+    super(
+        EndpointMetadata.get(ROUTE)
+            .operationId("getEth1Data")
+            .summary("Get new Eth1Data")
+            .description("Eth1Data we'd include in a new block if we are a proposer.")
+            .tags(TAG_TEKU)
+            .response(SC_OK, "Request successful", ETH1DATA_RESPONSE_TYPE)
+            .build());
+    this.chainDataProvider = dataProvider.getChainDataProvider();
+    this.eth1DataCache = eth1DataCache;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get new Eth1Data",
+      description = "Eth1Data we'd include in a new block if we are a proposer.",
+      tags = {TAG_TEKU},
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = @OpenApiContent(from = GetEth1DataResponse.class)),
+        @OpenApiResponse(status = RES_NOT_FOUND),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR)
+      })
+  @Override
+  public void handle(@NotNull final Context ctx) throws Exception {
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(final RestApiRequest request) throws JsonProcessingException {
+    request.respondAsync(
+        chainDataProvider
+            .getBeaconStateAtHead()
+            .thenApply(
+                mayBeState -> {
+                  if (mayBeState.isEmpty()) {
+                    return AsyncApiResponse.respondNotFound();
+                  } else {
+                    StateAndMetaData stateAndMetaData = mayBeState.get();
+                    Eth1Data eth1Vote = eth1DataCache.getEth1Vote(stateAndMetaData.getData());
```

Comment:
these are nit-picking,
 - variables should be final...
 - we're getting StateAndMetadata but looks like we want BeaconState, so make the variable BeaconState
 - if you do that, then it probably becomes simple enough that you could:
 ```
final BeaconState state = maybeState.get().getData();
return AsyncApiResponse.respondOk(eth1DataCache.getEth1Votes(state));
```
- i guess from that, 'maybe' is all lowercase, not capital b B
potentially make this even simpler, if by design Eth1DataCache can have `getEth1Votes(maybeStateAndMetadata)`
and then the logic is out of the handler...
so then you would just be doing
`thenApply(eth1DataCache::getEth1Vote);`